### PR TITLE
Extract Test-Specific External Command

### PR DESCRIPTION
**Pull request by mccurdyc**
## Proposed Changes
This pull request removes all test-specific code (i.e., the code for running `go test` with the `-coverprofile` flag always enabled). Now, neighbor is a tool that can run an arbitrary command on every project cloned during execution.

+ updated the documentation to reflect these changes
+ move test-specific code for `go test` with the `-coverprofile` always enabled to https://github.com/mccurdyc/go-cover (this repository still needs updated and cleaned up)

## TODO
+ add a `setup_external_command` to the config.json so that a user can specify a command to be run prior to cloning projects so that certain things don't need to occur on a per-project basis (e.g., swapping `go` to be a custom `go` with the `-coverprofile` flag always enabled.)

### CODE REVIEW

File: pkg/external/run.go [hunk 1]
```diff
@@ -14,24 +14,20 @@ import (
...
+			ctx.Logger.Infof("running external command on %s", p.Name)
 			err := os.Chdir(p.Directory)
 			if err != nil {
 				ctx.Logger.Error(err)
 				continue
 			}
 
-			if len(ctx.TestCmd) < 1 {
+			if len(ctx.ExternalCmd) < 1 {
 				ctx.Logger.Errorf("test command cannot be empty")
```
Comment (Michionlion) [inline pkg/external/run.go]: Should print "external command cannot be empty"

File: pkg/external/run.go [hunk 2]
```diff
@@ -53,28 +49,21 @@ func RunTests(ctx *neighbor.Ctx, ch <-chan github.ExternalProject) {
 				ctx.Logger.Errorf("failed to run test command with error %+v", err)
```
Comment (Michionlion) [inline pkg/external/run.go]: Should print "failed to run external command [...]"

File: pkg/neighbor/cmd.go [hunk 3]
```diff
@@ -6,7 +6,7 @@ import "strings"
 // external
 // internal
 
-// SetTestCmd sets the test command that will be run on external projects.
-func SetTestCmd(c *Ctx) {
-	c.TestCmd = strings.Split(c.Config.Contents.TestCmdStr, " ")
+// SetExternalCmd sets the command that will be run on external projects.
+func SetExternalCmd(c *Ctx) {
+	c.ExternalCmd = strings.Split(c.Config.Contents.ExternalCmdStr, " ")
```
Comment (Michionlion) [inline pkg/neighbor/cmd.go]: This won't handle calls that contain arguments denoted with `"`. We should note this in documentation or support it.

Comment (mccurdyc) [reply pkg/neighbor/cmd.go]: it should. I can confirm. It should just split on an empty space. I don't think that the quotes are part of the split string

**Pull request merged**